src/visual.ts

```diff
@@ -2822,6 +2822,7 @@ module powerbi.extensibility.visual {
 
             const xScale: any = this.xAxisProperties.scale,
                 yScale: any = this.yAxisProperties.scale,
+                viewport = this.viewport,
```

Comment:
Context of 'this' within 'each' was element and we were getting viewport == undefined